src/ziggurat/messaging/consumer.clj

```diff
@@ -116,15 +112,15 @@
       (doall (for [_ (range count)]
                (lb/get ch queue-name true))))))
 
-(defn- message-handler [wrapped-mapper-fn topic-entity]
+(defn- message-handler [wrapped-mapper-fn topic-entity ziggurat-channel-key]
   (fn [ch meta ^bytes payload]
-    (clog/with-logging-context {:consumer-group topic-entity} (process-message-from-queue ch meta payload topic-entity wrapped-mapper-fn))))
+    (clog/with-logging-context {:consumer-group topic-entity} (process-message-from-queue ch meta payload topic-entity wrapped-mapper-fn ziggurat-channel-key))))
 
-(defn- start-subscriber* [ch prefetch-count queue-name wrapped-mapper-fn topic-entity]
+(defn- start-subscriber* [ch prefetch-count queue-name wrapped-mapper-fn topic-entity ziggurat-channel-key]
```

Comment:
ziggurat-channel-key is added to this method for distinguishing between the caller. If channel-key is passed, channel-subscriber is the caller. If not, it's the retry-subscriber